prime-router/src/main/kotlin/transport/RESTTransport.kt

```diff
@@ -119,9 +124,27 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
                         logger.info("Token successfully added!")
 
                         // post the report
-                        val reportClient = httpClient ?: createDefaultHttpClient(
-                            jksCredential,
-                            accessToken,
+                        // If encryption is needed.
+                        val encryptClient = httpClient ?: createDefaultHttpClient(
+                            jksCredential, accessToken,
+                            restTransportInfo
+                        )
+                        val reportContent = if (restTransportInfo.encryptionKeyUrl.isNotEmpty()) {
+                            encryptTheReport(
```

Comment:
Why do you need a separate reportContent variable here? You should be able to re-use the reportContentOrg var